tests/test_tempfile.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+from typing import AnyStr
+
+import pytest
+
+from anyio import (
+    NamedTemporaryFile,
+    SpooledTemporaryFile,
+    TemporaryDirectory,
+    TemporaryFile,
+    gettempdir,
+    gettempdirb,
+    mkdtemp,
+    mkstemp,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+class TestTemporaryFile:
+    async def test_temporary_file(self) -> None:
+        data = b"temporary file data"
+        async with TemporaryFile[bytes]() as af:
+            await af.write(data)
+            await af.seek(0)
+            result = await af.read()
+
+        assert result == data
+        assert af.closed
+
+
+class TestNamedTemporaryFile:
+    async def test_named_temporary_file(self) -> None:
+        data = b"named temporary file data"
+        async with NamedTemporaryFile[bytes]() as af:
+            filename = af.name
+            assert os.path.exists(filename)  # type: ignore[arg-type]
+
+            await af.write(data)
+            await af.seek(0)
+            assert await af.read() == data
+
+        assert not os.path.exists(filename)  # type: ignore[arg-type]
+
+    async def test_exception_handling(self) -> None:
+        async with NamedTemporaryFile[bytes]() as af:
+            filename = af.name
+            assert os.path.exists(filename)  # type: ignore[arg-type]
```

Comment:
Why is this a mypy error?

Reply:
Because the `name` attribute is retrieved using `__getattr__()` which is declared to return `object`. I think this has been used elsewhere in anyio too – perhaps we should declare it to return `Any` instead?

Reply:
Of course, that won't be necessary if this is just reimplemented by copying stdlib code.